tests/queries/0_stateless/03444_user_defined_types_complex.sql

```diff
@@ -0,0 +1,29 @@
+DROP TYPE IF EXISTS UserId;
+DROP TYPE IF EXISTS Email;
+DROP TYPE IF EXISTS Phone;
+DROP TYPE IF EXISTS ComplexArray;
+DROP TYPE IF EXISTS ParameterizedType;
+
+CREATE TYPE UserId AS UInt64 
+    INPUT 'toUInt64(assumeNotNull(value))' 
+    OUTPUT 'toString(value)' 
+    DEFAULT '0';
+SHOW TYPE UserId;
+
+CREATE TYPE ComplexArray AS Array(Tuple(String, UInt64));
+SHOW TYPE ComplexArray;
+
+CREATE TYPE ParameterizedType(T, U) AS Tuple(T, Array(U));
```

Comment:
So a user-defined type is basically a type alias. Can you please add some documentation about it?

Reply:
https://www.postgresql.org/docs/current/sql-createdomain.html

PostgreSQL also support CREATE TYPE, but it need library with implementation in c. 

One usecase: https://github.com/ClickHouse/ClickHouse/issues/56711

Reply:
Biggest question here, is:
Currently, if you use UDF in table creation, it automatically being unwrapped to it's definition. 
It makes it bit more strange/useless, but still tolerable with UDF. But this approach wouldn't work with types.